### PR TITLE
port tests for F.erase and transforms.RandomErasing

### PR DESCRIPTION
**Pull request by pmeier**
cc @vfdev-5

### CODE REVIEW

File: test/test_transforms_v2_refactored.py [hunk 1]
```diff
@@ -2485,3 +2485,145 @@ def test_correctness(self):
...
+            transforms.RandomErasing(scale=[-1, 2])
+
+        transform = transforms.RandomErasing(value=[1, 2, 3, 4])
+
+        with pytest.raises(ValueError, match="If value is a sequence, it should have either a single value"):
+            transform._get_params([make_image()])
+
+    @pytest.mark.parametrize(
+        "make_input",
+        [make_bounding_boxes, make_detection_mask, make_segmentation_mask],
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:2618]: Do we need to parametrize over both mask types?

File: test/test_transforms_v2_refactored.py [hunk 2]
```diff
@@ -2485,3 +2485,145 @@ def test_correctness(self):
...
+    def test_kernel_image(self, dtype, device):
+        check_kernel(F.erase_image, make_image(self.INPUT_SIZE, dtype=dtype, device=device), **self.FUNCTIONAL_KWARGS)
+
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_kernel_image_inplace(self, dtype, device):
+        input = make_image(self.INPUT_SIZE, dtype=dtype, device=device)
+        input_version = input._version
+
+        output_out_of_place = F.erase_image(input, **self.FUNCTIONAL_KWARGS)
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:2507]: We also need to check the Python objects, i,e. we need to add the following checks:

```py

        assert output_out_of_place is not input
...

        assert output_inplace is input

```

**Review by NicolasHug — APPROVED** [review]
Thanks Philip, LGTM if green

File: test/test_transforms_v2_refactored.py (hunk 2 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:2507]: I'll put them in for clarity, but I think these checks are redundant or potentially too strict:

- I'm not aware of a way to change the storage of an existing Python Tensor object. Meaning, it is safe to assume that if the storage pointer mismatches, we also have different Python objects
- I'm not sure if `inplace` refers to the Python object or the underlying storage. If it just applies to the storage `assert output_inplace is input` could actually fail:

    ```python
    import torch
    
    a = torch.rand(5)
    
    b = a.add_(1)
    
    assert b.data_ptr() == a.data_ptr()
    assert b is a
    
    c = b.view(-1)
    
    assert c.data_ptr() == a.data_ptr()
    assert c is not a
    ```

File: test/test_transforms_v2_refactored.py (hunk 1 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:2618]: Nope, not for this check. For kernel checks we need to due to their different nature.

File: test/test_transforms_v2_refactored.py (hunk 2 above)
Comment (NicolasHug) [reply test/test_transforms_v2_refactored.py:2507]: > I'm not sure if inplace refers to the Python object or the underlying storage

Both. "inplace" means users are allowed to do

```py
F.erase_image(input, **self.FUNCTIONAL_KWARGS)
# use input now
```

without worrying about a return value. It's the same as for `inplace_()` operators. So it's best to enforce that contract in the tests as well.
